drivers/leds/leds-ktd202x.c

```diff
@@ -0,0 +1,416 @@
+	num_steps = (*delay_on + *delay_off - KTD202X_TIME_MIN) /
+		    KTD202X_TIME_STEP + 1;
+	num_steps = min(126, num_steps);
+	on = (*delay_on * KTD202X_ON_MAX) / (*delay_on + *delay_off);
+
+	delay_total = num_steps * KTD202X_TIME_STEP + KTD202X_TIME_MIN;
+	*delay_on = (delay_total * on) / KTD202X_ON_MAX;
+	*delay_off = delay_total - *delay_on;
+
+	/* Set timings */
+	ret = regmap_write(led->chip->regmap, KTD202X_REG_FLASH_PERIOD,
+			   num_steps);
+	if (ret)
+		goto out;
+	ret = regmap_write(led->chip->regmap, KTD202X_REG_PWM1_TIMER, on);
+	if (ret)
+		goto out;
+
+	ret = regmap_update_bits(led->chip->regmap, KTD202X_REG_CHANNEL_CTRL,
+				 KTD202X_CHANNEL_CTRL_MASK(num),
+				 KTD202X_CHANNEL_CTRL_PWM1(num));
+
+out:
+	mutex_unlock(&led->chip->mutex);
+
+	return ret;
+}
+
+static int ktd202x_probe_dt(struct ktd202x *chip)
+{
+	struct device_node *np = dev_of_node(&chip->client->dev), *child;
+	int count, ret, i = 0;
+	struct ktd202x_led *led;
+	const struct ktd202x_info *ktd202x;
+
+	ktd202x = device_get_match_data(&chip->client->dev);
+	if (!ktd202x)
+		return -ENODEV;
+
+	count = of_get_available_child_count(np);
+	if (!count || count > ktd202x->num_leds)
+		return -EINVAL;
+
+	regmap_write(chip->regmap, KTD202X_REG_RESET_CONTROL,
+		     KTD202X_RSTR_RESET);
+	udelay(200); /* allow the device to execute the complete reset */
+
+	for_each_available_child_of_node(np, child) {
+		struct led_init_data init_data = {};
+		u32 source;
+
+		ret = of_property_read_u32(child, "reg", &source);
+		if (ret != 0 || source >= ktd202x->num_leds) {
+			dev_err_probe(&chip->client->dev, ret,
+				      "Couldn't read LED address.\n");
+			count--;
+			continue;
+		}
+
+		led = &chip->leds[i];
+		led->num = source;
+		led->chip = chip;
+		init_data.fwnode = of_fwnode_handle(child);
+
+		led->cdev.brightness_set_blocking = ktd202x_brightness_set;
+		led->cdev.blink_set = ktd202x_blink_set;
+
+		ret = devm_led_classdev_register_ext(&chip->client->dev,
+						     &led->cdev, &init_data);
+		if (ret < 0) {
+			of_node_put(child);
+			return ret;
+		}
+
+		i++;
+	}
+
+	if (!count)
+		return -EINVAL;
+
+	chip->num_leds = ktd202x->num_leds;
+
+	return 0;
+}
+
+static const struct regmap_config ktd202x_regmap_config = {
+	.reg_bits = 8,
+	.val_bits = 8,
+	.max_register = 0x09,
+	.cache_type = REGCACHE_FLAT,
+	.reg_defaults = ktd202x_reg_defaults,
+	.num_reg_defaults = ARRAY_SIZE(ktd202x_reg_defaults),
+};
+
+static int ktd202x_probe(struct i2c_client *client)
+{
+	struct ktd202x *chip;
+	int ret;
+
+	chip = devm_kzalloc(&client->dev, sizeof(*chip), GFP_KERNEL);
+	if (!chip)
+		return -ENOMEM;
+
+	mutex_init(&chip->mutex);
+	mutex_lock(&chip->mutex);
+
+	chip->client = client;
+	i2c_set_clientdata(client, chip);
+
+	chip->regmap = devm_regmap_init_i2c(client, &ktd202x_regmap_config);
+	if (IS_ERR(chip->regmap)) {
+		ret = PTR_ERR(chip->regmap);
+		dev_err_probe(&client->dev, ret,
+			      "Failed to allocate register map.\n");
```

Comment:
if you'd think it would be more pretty this way, 
```suggestion
		ret = dev_err_probe(&client->dev, PTR_ERR(chip->regmap),
			      "Failed to allocate register map.\n");
```